### ISSUE TITLE
HPCToolkit and HPCtoolkit-externals releases 

### ISSUE THREAD

**Issue by giacomo-mcevoy** [HPCToolkit/hpctoolkit-externals#2]
This is more a question than an issue: will there be a 2017.10 release for the hpctoolkit-externals? I ask this because the HCPToolkit release 2017.10 cannot be compiled with the HPCToolkit-externals release 2017.06 on aarch64: it seems to be expecting a newer version of the libunwind library.

`../../../../src/tool/hpcrun/unwind/generic-libunwind/libunw-unwind.c: In function ‘hpcrun_unw_init_cursor’:
../../../../src/tool/hpcrun/unwind/generic-libunwind/libunw-unwind.c:190:3: warning: implicit declaration of function ‘unw_init_local2’ [-Wimplicit-function-declaration]
   if (ctx != NULL && unw_init_local2(unw_cursor, ctx, UNW_INIT_SIGNAL_FRAME) == 0) {
   ^
../../../../src/tool/hpcrun/unwind/generic-libunwind/libunw-unwind.c:190:55: error: ‘UNW_INIT_SIGNAL_FRAME’ undeclared (first use in this function)
   if (ctx != NULL && unw_init_local2(unw_cursor, ctx, UNW_INIT_SIGNAL_FRAME) == 0) {
                                                       ^
../../../../src/tool/hpcrun/unwind/generic-libunwind/libunw-unwind.c:190:55: note: each undeclared identifier is reported only once for each function it appears in
../../../../src/tool/hpcrun/unwind/generic-libunwind/libunw-unwind.c: In function ‘hpcrun_unw_step’:
../../../../src/tool/hpcrun/unwind/generic-libunwind/libunw-unwind.c:254:3: warning: implicit declaration of function ‘unw_apply_reg_state’ [-Wimplicit-function-declaration]
   unw_apply_reg_state(unw_cursor, uwi->recipe);
   ^
../../../../src/tool/hpcrun/unwind/generic-libunwind/libunw-unwind.c: In function ‘libunw_build_intervals’:
../../../../src/tool/hpcrun/unwind/generic-libunwind/libunw-unwind.c:305:28: error: ‘UNW_INIT_SIGNAL_FRAME’ undeclared (first use in this function)
   unw_init_local2(&c, &uc, UNW_INIT_SIGNAL_FRAME);
                            ^
../../../../src/tool/hpcrun/unwind/generic-libunwind/libunw-unwind.c:310:3: warning: implicit declaration of function ‘unw_reg_states_iterate’ [-Wimplicit-function-declaration]
   int status = unw_reg_states_iterate(&c, dwarf_reg_states_callback, &b);
`

hpcrun is looking for UNW_INIT_SIGNAL_FRAME, but it is not declared in the libunwind from hpctoolkit-externals 2017.06, only on the latest version. Naturally, it was possible to update libunwind manually with libunwind-2017-10-02.tar.bz2, this worked. 

Thought I would leave this comment to note that the latest releases cannot be used together in aarch64.

**Comment by giacomo-mcevoy** [comment]
Update on the previous post: On aarch64, HPCToolkit 2017.10 can be compiled with HPCToolkit-externals 2017.06 + libunwind-2017-10-02.tar.bz2, but it does not work. The following runtime error is thrown when using hpcrun:

`symbol lookup error: /opt/bullxde/profilers/hpctoolkit/2017.10/lib/hpctoolkit/libhpcrun.so: undefined symbol: _ULaarch64_init_local2`

Reverted to using HPCToolkit 2017.06 + HPCToolkit-externals 2017.06, seems to be working fine sofar.

**Comment by mwkrentel** [comment]
First, I'd suggest sending questions like this to the hpctoolkit-forum
mailing list.  That's where you'll get someone's attention.

If it's more of a to-do item/bug report and you want to use git
issues, then put the issue on hpctoolkit, not externals.
I wasn't really expecting anything to show up here.

I looked at what's on github.  When we made the release in October,
we forgot to make a tag in the externals repo.  I made a configure
package version for extenals, but forgot to make a tag.

As you've probably figured out, if you use externals from the same
date as the hpctooklit release, it should work.  It probably works to
use today's externals with the toolkit release, but that's not
guaranteed.

Anyway, we're working toward a new release in the next couple weeks,
and I'll try to remember to tag both then.

So, unless there's an actual bug other than forgetting the tag,
I'll close this issue.
